src/rest.js

```diff
@@ -14,53 +14,50 @@
  * limitations under the License.
  */
 
-var cachedResponseLastRefresh = null;
-var cachedResponse = null;
-
 function sql(datasetKey, sqlQuery, lastRefresh, sampleExtraction) {
-    if (!(lastRefresh && cachedResponseLastRefresh) ||
-        (lastRefresh > cachedResponseLastRefresh)) {
-
-        var params = {
-            method: 'post',
-            headers: {
-                'Accept': 'application/x-ndjson',
-                'Authorization': 'Bearer ' + getOAuthService().getAccessToken()
-            },
-            payload: {
-                'query': sqlQuery
-            },
-            muteHttpExceptions: true
-        };
+    var params = {
+        method: 'post',
+        headers: {
+            'Accept': 'application/x-ndjson',
+            'Authorization': 'Bearer ' + getOAuthService().getAccessToken()
+        },
+        payload: {
+            'query': sqlQuery
+        },
+        muteHttpExceptions: true
+    };
 
-        var response = UrlFetchApp.fetch(getSqlEndpoint(datasetKey), params);
-        if (response.getResponseCode() === 200) {
-            cachedResponse = response;
-            cachedResponseLastRefresh = lastRefresh;
+    var success = false;
+    var lock = LockService.getUserLock();
+    success = lock.tryLock(60000);
+    
+    var response = UrlFetchApp.fetch(getSqlEndpoint(datasetKey), params);
```

Comment:
if success is false, won't this still get executed?

Reply:
Yes, I will add code to throw an error
` "Server Busy please try after some time"`